casper/src/main/scala/coop/rchain/casper/api/BlockReportAPI.scala

```diff
@@ -43,56 +43,40 @@ class BlockReportAPI[F[_]: Concurrent: Metrics: EngineCell: Log: SafetyOracle: B
 ) {
   implicit val source                                       = Metrics.Source(CasperMetricsSource, "report-replay")
   val blockLockMap: TrieMap[BlockHash, MetricsSemaphore[F]] = TrieMap.empty
-  private def replayBlock(
-      b: BlockMessage
-  )(implicit casper: MultiParentCasper[F]) =
+
+  private def replayBlock(b: BlockMessage)(implicit casper: MultiParentCasper[F]) =
     for {
       reportResult <- reportingCasper.trace(b)
       lightBlock   <- BlockAPI.getLightBlockInfo[F](b)
       deploys      = createDeployReport(reportResult.deployReportResult)
       sysDeploys   = createSystemDeployReport(reportResult.systemDeployReportResult)
       blockEvent   = BlockEventInfo(Some(lightBlock), deploys, sysDeploys, reportResult.postStateHash)
-    } yield blockEvent.asRight[Error]
+    } yield blockEvent
 
-  private def blockReportWithinLock(
-      forceReplay: Boolean,
-      b: BlockMessage
-  )(implicit casper: MultiParentCasper[F]) =
+  private def blockReportWithinLock(forceReplay: Boolean, b: BlockMessage)(
+      implicit casper: MultiParentCasper[F]
+  ) =
     for {
       semaphore <- MetricsSemaphore.single
-      lock = blockLockMap
-        .getOrElseUpdate(b.blockHash, semaphore)
+      lock      = blockLockMap.getOrElseUpdate(b.blockHash, semaphore)
       result <- lock.withPermit(
                  for {
                    cached <- reportStore.get(b.blockHash)
                    res <- if (cached.isEmpty || forceReplay)
-                           for {
-                             blockEvent <- replayBlock(b)
-                             _ <- blockEvent.fold(
-                                   _ => ().pure[F],
-                                   data => reportStore.put(b.blockHash, data)
```

Comment:
@tgrospic The result would never be stored in cache because  `reportStore.put(b.blockHash, data)` is removed now.